pkg/azure/blob.go

```diff
@@ -343,6 +356,18 @@ func sharedCredentialFromSecret(endpoint string, secret *corev1.Secret) (*azblob
 	return nil, nil
 }
 
+// sasTokenFromSecret retrieves the SAS Token from the `sasKey`. It returns an empty string if the Secret
+// does not contain a valid set of credentials.
+func sasTokenFromSecret(secret *corev1.Secret) (string, error) {
+	if sasKey, hasSASKey := secret.Data[sasKeyField]; hasSASKey {
```

Comment:
We may want to go a step further, and ensure that the required query parameters were been provided.
From their examples those seem to be:

```
signedstart=2009-02-09  
signedexpiry=2009-02-10  
signedresource=c  
signedpermissions=r  
signature=dD80ihBh5jfNpymO5Hg1IdiJIEvHcJpCMiCMnN/RnbI=  
signedidentifier=YWJjZGVmZw==  
signedversion=2012-02-12
```

https://docs.microsoft.com/en-us/rest/api/storageservices/service-sas-examples

Reply:
Docs is slightly confusing. The actual SAS token generated in the Azure Portal has none of these fields. It looks more like this:
```
?sv=2013-08-15&st=2013-08-16&se=2013-08-17&sr=c&sp=r&rscd=file;%20attachment&rsct=binary &sig=YWJjZGVmZw%3d%3d&sig=a39%2BYozJhGp6miujGymjRpN8tsrQfLo9Z3i8IRyIpnQ%3d HTTP/1.1  
```
 I will look through the documentation again on which of the fields is actually required and which is optional

Reply:
Let's not go there, MSFT could change the fields at any time and we can't keep track of this. We should validate the URL and if the fields are wrong then it's up the user to fix them.